lib/ansible/modules/cloud/azure/azure_rm_aci.py

```diff
@@ -0,0 +1,471 @@
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            ip_address=dict(
+                type='str',
+                required=False,
+                default=None,
+                choices=['public']
+            ),
+            ports=dict(
+                type='list',
+                required=False
+            ),
+            registry_login_server=dict(
+                type='str',
+                required=False,
+                default=None
+            ),
+            registry_username=dict(
+                type='str',
+                required=False,
+                default=None
+            ),
+            registry_password=dict(
+                type='str',
+                required=False,
+                default=None
+            ),
+            containers=dict(
+                type='list',
+                required=True
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.tags = None
+        self.state = None
+        self.ip_address = None
+
+        self.containers = None
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+
+        super(AzureRMContainerInstance, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                       supports_check_mode=True,
+                                                       supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        response = None
+        results = dict()
+        to_be_updated = False
+
+        self.mgmt_client = self.get_mgmt_svc_client(ContainerInstanceManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('resource group {} not found'.format(self.resource_group))
+        if not self.location:
+            self.location = resource_group.location
+
+        response = self.get_aci()
+
+        if not response:
+            self.log("Container Group doesn't exist")
+
+            if self.state == 'absent':
+                self.log("Nothing to delete")
+            else:
+                to_be_updated = True
+        else:
+            self.log("Container instance already exists")
+
+            if self.state == 'absent':
+                self.delete_aci()
+                self.results['changed'] = True
+                self.log("ACI instance deleted")
+            elif self.state == 'present':
+                self.log("Need to check if container group has to be deleted or may be updated")
```

Comment:
Ideally we'd also actually check at least the basic "shape" of the specified containers as part of the change detection (assuming this resource type supports update, which the REST API and SDK appear to). This is always something that could be added later too, but nice to have if the underlying resource really does support changes.

Reply:
actually in case of container instance, it's not possible to update anything. any attempt to update existing container just fails. it's difficult to judge based on passed parameters whether container instance should be updated or not, for example user may want to update container instance because image has changed. well, in such case they could use tags do detect that, but there are a few other things which make it a bit tricky...